mmv1/third_party/terraform/.teamcity/components/sweepers.kt

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import jetbrains.buildServer.configs.kotlin.BuildType
+import jetbrains.buildServer.configs.kotlin.AbsoluteId
+
+class sweeperBuildConfigs() {
+
+    fun preSweeperBuildConfig(path: String, manualVcsRoot: AbsoluteId, parallelism: Int, environmentVariables: ClientConfiguration) : BuildType {
+        val testPrefix = "TestAcc"
+        val testTimeout = "12"
+        val sweeperRegions = "us-central1"
+        val sweeperRun = "" // Empty string means all sweepers run
+
+        val configName = "Pre-Sweeper"
+        val sweeperStepName = "Pre-Sweeper"
+
+        return createBuildConfig(manualVcsRoot, preSweeperBuildConfigId, configName, sweeperStepName, parallelism, testPrefix, testTimeout, sweeperRegions, sweeperRun, path, environmentVariables)
+   }
+
+    fun postSweeperBuildConfig(path: String, manualVcsRoot: AbsoluteId, parallelism: Int, triggerConfig: NightlyTriggerConfiguration, environmentVariables: ClientConfiguration, dependencies: ArrayList<String>) : BuildType {
+        val testPrefix = "TestAcc"
+        val testTimeout = "12"
+        val sweeperRegions = "us-central1"
+        val sweeperRun = "" // Empty string means all sweepers run
+
+        val configName = "Post-Sweeper"
+        val sweeperStepName = "Post-Sweeper"
+
+        val build = createBuildConfig(manualVcsRoot, postSweeperBuildConfigId, configName, sweeperStepName, parallelism, testPrefix, testTimeout, sweeperRegions, sweeperRun, path, environmentVariables)
+
+        build.addDependencies(dependencies) // Make dependent on package builds before starting
+        build.addTrigger(triggerConfig)     // Post-Sweeper is triggered by cron, and dependencies result in other builds being queued
```

Comment:
Note the the post-sweeper is the only part of the config at the moment that is triggered by cron.

When the post-sweeper is triggered it will cause each of the package + service package builds to be triggered, due to dependencies set on the post-sweeper build.

Each package + service package build is dependent on the pre-sweeper. They also trigger the pre-sweeper build to be queued due to a dependency.